### ISSUE TITLE
Unable open files created with nightly version of xournal

### ISSUE THREAD

**Issue by MichelReims** [xournalpp/xournalpp#4127]
I work on windows 10. 
I created files with a January 2021 version of xournal. 
I can no longer open them with the xournalpp-1.1.1+dev-nightly.20220629-windows.exe version, but I can open them with the xournalpp- 1.1.1-windows.exe
Below an example of file
[Correction TD 1 (Exos 1-5).zip](https://github.com/xournalpp/xournalpp/files/9032577/Correction.TD.1.Exos.1-5.zip)

**Comment by rolandlo** [comment]
Thanks for reporting. Do you have any closer estimate on what version was the last one that worked?

**Comment by MichelReims** [comment]
No, I don't know; they are old files and I don't remember when the were created

**Comment by Technius** [comment]
Using this file, got a crash on latest master on Linux. Segfault here:
https://github.com/xournalpp/xournalpp/blob/9ae2429bffd9906ee3e132242b3a35df3ff7b7c4/src/core/view/DocumentView.cpp#L65

Bisect culprit is `8d07503`. @bhennion, could you take a look?

**Comment by Technius** [comment]
This is printed in the logs before the crash happens:
```
** (xournalpp:28599): WARNING **: 01:18:25.493: BackgroundView::createForPage for 'Copy' page type
```
I believe an unhandled case in `BackgroundView::createForPage` is being triggered:
https://github.com/xournalpp/xournalpp/blob/9ae2429bffd9906ee3e132242b3a35df3ff7b7c4/src/core/view/background/BackgroundView.cpp#L80-L82

**Comment by bhennion** [comment]
Hmm... I'll have a look. Thanks @technius for investigating!

Do you know what the `PageTypeFormat::Copy` is supposed to stand for? I assumed it was for internal use only, to state that we were copying the background from another page. I did not think it could ever land in the .xopp file.

For instance, when creating a new page with "Copy current element" selected in the drop down menu does not create a page of type Copy, but indeed copies the the current page background.

**Comment by MichelReims** [comment]
I created this file with a September 2020 nightly version of xournal,  but can't remember which one.
For the background, I used the following actions : "Journal->Paper Background->Plain",  then "Apply to all pages"

**Comment by bhennion** [comment]
Thanks! I now understand: there was a bug that made the "Apply [background] to all pages" menu entry assigning the type Copy to pages. This produced buggy .xopp files (look for `style=":copy"` in the xml file). This bug was fixed by @Technius in 257b935b.
Since 8d07503, the unwanted behaviour is caught by the rendering system, and it is no longer possible to open those buggy files.

I think the solution is to fix those buggy files upon opening.

**Comment by Technius** [comment]
Thanks for investigating, @bhennion. The backwards compatibility snippet sounds good to me. I would also add a check to the SaveHandler to emit a warning if the copy format is being saved.

I think I found why the copy format is being produced. Here's a simpler crash reproducer:

1. New document with 2 pages.
2. Use the page format dialog in the toolbar (not the menu) to set page type to copy.
3. Apply page format to current page.
4. Observe segfault.

The issue is in `PageBackgroundChangeController::commitPageTypeChange()`, which does not seem to have the correct handling for the copy page type.

Since the issues are somewhat orthogonal, I think we should split these fixes into two parts: 1) backwards compatibility snippet; 2) fixing the page copy logic. I can confirm that the problem (2) also affects `release-1.1`, so we will need to backport the corresponding fix.

**Comment by bhennion** [comment]
> Since the issues are somewhat orthogonal, I think we should split these fixes into two parts: 1) backwards compatibility snippet; 2) fixing the page copy logic. I can confirm that the problem (2) also affects `release-1.1`, so we will need to backport the corresponding fix.

First, nice catch!
I agree this is a somewhat separated issue. Maybe open (yet) another ticket.

In the meantime, I proposed a fix for issue (1) in #4139.
